ambry-frontend/src/main/java/com/github/ambry/frontend/PostBlobHandler.java

```diff
@@ -322,6 +322,16 @@ private void setSignedIdMetadataAndBlobSize(BlobProperties blobProperties) throw
             RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SESSION, true));
         metadata.put(EXPIRATION_TIME_MS_KEY,
             Long.toString(Utils.addSecondsToEpochTime(time.milliseconds(), blobProperties.getTimeToLiveInSeconds())));
+        if (blobProperties.getReservedMetadataBlobId() != null) {
+          metadata.put(RestUtils.Headers.RESERVED_METADATA_ID, blobProperties.getReservedMetadataBlobId());
+        } else {
+          ReservedMetadataIdMetrics.getReservedMetadataIdMetrics(
+              frontendMetrics.getMetricRegistry()).noReservedMetadataFoundForChunkedUploadResponseCount.inc();
```

Comment:
QQ (Not related to this PR). For other metrics classes like RouterMetrics, FrontendMetrics, etc, we seem to create them in the factory classes. Would it involve lot of code changes to do similar for `ReservedMetadataIdMetrics` also? For example, would we have to create it at the beginning in `RestServer.java`, etc and pass it to all components?